internal/set/set.go

```diff
@@ -44,3 +44,7 @@ func (s *Set[E]) Empty() bool {
 func (s *Set[E]) Values() []E {
 	return maps.Keys(s.m)
 }
+
+func (s Set[E]) Size() int {
```

Comment:
```suggestion
func (s *Set[E]) Size() int {
```
please.

The current version works just fine, but I managed to inattentively screw up the `Add`/`Delete` methods (https://github.com/containers/image/pull/1951 ), so using a pointer receiver everywhere would hopefully make it harder for me to make the same mistake again.

Reply:
Changed it.